test/jdk/java/nio/file/Files/Links.java

```diff
@@ -131,6 +131,42 @@ static void testSymLinks(Path dir) throws IOException {
             Files.deleteIfExists(mydir);
             Files.deleteIfExists(link);
         }
+
+        // Check message of NotLinkException
```

Comment:
It might be that checking the message for just one single non-link, instead of both a file and a directory, is sufficient.